bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/AbstractComponent.java

```diff
@@ -94,14 +115,23 @@ public AbstractComponent(ComponentFactory.ComponentConfiguration componentConfig
 
         String name = channelConfiguration.getName();
         if (name != null && !name.isEmpty()) {
-            String groupId = this.haID.getGroupId(channelConfiguration.getUniqueId());
+            groupId = this.haID.getGroupId(channelConfiguration.getUniqueId(), newStyleChannels);
 
-            this.channelGroupTypeUID = new ChannelGroupTypeUID(MqttBindingConstants.BINDING_ID, groupId);
-            this.channelGroupUID = new ChannelGroupUID(componentConfiguration.getThingUID(), groupId);
+            if (this.singleChannelComponent) {
+                this.channelGroupUID = null;
+            } else {
+
+                this.channelGroupUID = new ChannelGroupUID(componentConfiguration.getThingUID(), groupId);
+            }
         } else {
-            this.channelGroupTypeUID = null;
+            if (newStyleChannels && this.singleChannelComponent) {
+                groupId = haID.component;
+            } else {
+                groupId = "";
```

Comment:
Should not it be null as in the org.openhab.core.thing.ChannelUID.getGroupId?

Reply:
It can't be null, since it's not just used as part of a channel id, but also for other uses in the thing handler that don't allow null.